models/repo_pushmirror.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"errors"
+	"time"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+var (
+	// ErrPushMirrorNotExist mirror does not exist error
+	ErrPushMirrorNotExist = errors.New("PushMirror does not exist")
+)
+
+// PushMirror represents mirror information of a repository.
+type PushMirror struct {
+	ID         int64       `xorm:"pk autoincr"`
+	RepoID     int64       `xorm:"INDEX"`
+	Repo       *Repository `xorm:"-"`
+	RemoteName string
+
+	Interval       time.Duration
+	CreatedUnix    timeutil.TimeStamp `xorm:"created"`
+	LastUpdateUnix timeutil.TimeStamp `xorm:"INDEX last_update"`
+	LastError      string
+}
+
+// AfterLoad is invoked from XORM after setting the values of all fields of this object.
+func (m *PushMirror) AfterLoad(session *xorm.Session) {
```

Comment:
I'm not a fan of load it each time intransparent, we should create a "LoadRepo" func for *PushMirror* struct
And exec it where needed, this way we can also pass the error upwards and not just drop it into the logs

Reply:
That was copied from the pull mirror. While I agree with your intention there is a fine line between being intransparent and cluttering the code every time the object is used. For the mirror structs for example the load of the repository is needed all the time. Maybe it would be useful if `AfterLoad` could return an error which would be propagated to the caller of `Get`, `Find` and `Iterate`.

Reply:
hmm an xorm feature request ...

and it looks like the "pull mirror" code need some tuning ...